Automattic/wp-calypso
# Themes: Use PageViewTracker, drop ClientSideEffects

Use `PageViewTracker` to track page views for themes sheets (which previously weren't tracked at all), and use it to replace `ClientSideEffects` for `/design`. 

Also, add `/type/:tier` to `/design`'s page view path.

To test:
* In the console: type `localStorage.setItem( 'debug', 'calypso:analytics*' );`
* Land on a theme details page (try directly, and via a theme's 'Details' link). Verify that an event like `Recording Page View ~ [URL: /themes/owari/:slug] [Title: Themes > Details Sheet]` is tracked.

Supersedes #3420, and fixes #4566.

Test live: https://calypso.live/?branch=update/theme-sheet-track-page-views
